Source/MultipartFormData.swift

```diff
@@ -455,9 +455,11 @@ open class MultipartFormData {
         inputStream.open()
         defer { inputStream.close() }
 
-        while inputStream.hasBytesAvailable {
-            var buffer = [UInt8](repeating: 0, count: streamBufferSize)
-            let bytesRead = inputStream.read(&buffer, maxLength: streamBufferSize)
+        var bytesLeftToRead = bodyPart.bodyContentLength
+        while inputStream.hasBytesAvailable && bytesLeftToRead > 0 {
+            let bufferSize = min(streamBufferSize, Int(bytesLeftToRead))
```

Comment:
1. This prevents reading more than needed;  AND
2. Prevents math overflow when subtracting `bytesRead ` from `bytesLeftToRead `